build/Builder.js

```diff
@@ -11,20 +13,22 @@ class Builder{
 	}
 
 	getVariables(){
-		let varString = Object.keys(this.vars).map((k, i) => `$${k}: ${this.vars[k]};` ).join("\n");
+		const varString = Object.entries(this.vars).map((k, v) => `$${k}: ${v};`).join('\n');
+
 		console.log(`Vars: ${varString}`);
 		return varString;
 	}
 
 	build(src, dest){
 		src = path.resolve(src);
-		let dir = path.dirname(src);
-		let res = sass.compileString(this.getVariables() + `@import '${src}';`, {
-			loadPaths: [dir]
+		const dir = path.dirname(src);
+		const res = sass.compileString(this.getVariables() + `@import '${src}';`, {
+			loadPaths: [ dir ]
```

Comment:
Assuming all the `const`s are after passing through the linter?

I'm not really a fan of putting the spaces into the array declaration to be honest, it feels weird?

Reply:
`const` in javascript refers to a variable that can not be explicitly defined after initialisation. It is normally best practice to initialise all variables that don't change as `const` to catch any errors if you might redefine something accidentally.

If we need to reassign a value we should use `let` but the default is assumed to be const.

Reply:
I can change the lint rule for that if you want. There was an object spacing rule I liked that looked like this:

```
const a = { b: 'c' };
--- instead of ---
const a = {b: 'c'};
```

I prefered the former so I matched the array bracket spacing rule, but if we want to change that then it's easily doable.

Reply:
Perhaps `let` is just me being lazy then, as I'm never sure if I'm gonna write back to that var and it's a pain having to track it down and change it 😅

For me personally, I would prefer to define arrays as

```
const arr = [1, 2, 3]
```

and objects as multiline declarations

```
const obj = {
                         A: 1,
                         B: 7,
                         C: "purple"
                      }
```


(Excuse formatting, I'm on my phone, but you get the vibe)

Reply:
Oh my word that obj was horrible, it was supposed to be one more indent in 😅

Reply:
That can certainly be arranged; it depends how opinionated we want to go.

I can scrap the arr rule (or invert it to require it to be build like your example?)
And [there is a rule](https://eslint.org/docs/rules/object-curly-newline) I can add to enforce objects to be declared as multilines?

Reply:
Again I'm not convinced that this type of linting gives us any advantage? It might just be me used to not having a linter though...

I understand the purpose of a linter is to ensure code styles, but really if you look at `[1, 2, 3]` then you'll understand `[ 1, 2, 3 ]`?

For objects it does look cleaner split over multiple lines imo - you are more likely to notice an object declaration as 
```
const obj = {
    a: 1,
    b: 2,
    c: "purple"
};
```

than

```
const obj = { a:1, b:2, c: "purple" };
```

but again I'm not sure if going to that level of detail is really necessary **so long as the object is properly declared**

Reply:
In that case then, I will go through and refine the ruleset to best practices and error prevention as opposed to general code style - Thinking about it now I agree with you on that it doesn't really matter.

I'll keep this PR on hold then until I've got the new ruleset merged into BritishYouthBandAssociation/eslint-config and we'll revisit this.